marklogic-data-hub/src/main/java/com/marklogic/hub/impl/EntityManagerImpl.java

```diff
@@ -128,15 +136,23 @@ private boolean deployQueryOptions(DatabaseClient client, String filename) {
 
     @Override public boolean saveDbIndexes() {
         try {
-            Path dir = hubConfig.getEntityDatabaseDir();
+    		Path dir = hubConfig.getEntityDatabaseDir();
+        	File finalFile = Paths.get(dir.toString(), HubConfig.FINAL_ENTITY_DATABASE_FILE).toFile();
+        	File stagingFile = Paths.get(dir.toString(), HubConfig.STAGING_ENTITY_DATABASE_FILE).toFile();
             if (!dir.toFile().exists()) {
                 dir.toFile().mkdirs();
             }
-            File finalFile = Paths.get(dir.toString(), HubConfig.FINAL_ENTITY_DATABASE_FILE).toFile();
-            File stagingFile = Paths.get(dir.toString(), HubConfig.STAGING_ENTITY_DATABASE_FILE).toFile();
 
-            long lastModified = Math.max(finalFile.lastModified(), stagingFile.lastModified());
-            List<JsonNode> entities = getModifiedRawEntities(lastModified);
+            List<JsonNode> entities = getAllEntities();
+            List<String> databaseEntities = getEntityNamesInDatabase();
```

Comment:
You can remove this part and accompanying function.  just stick with all entities.